libs/frontend/feature-build/src/components/BuildDndOverlay.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+  <div class="build-dnd-overlay" :style="overlayStyle" />
+</template>
+
+<script lang="ts" setup>
+import { computed } from 'vue'
+import { runtimeContext } from '@pubstudio/frontend/util-runtime'
+import { useBuild } from '../lib/use-build'
+
+const hoverOutlineThickness = 6
+const outlineColor = '#000'
+const selfOutlineColor = '#F82389'
+
+const { editor } = useBuild()
+
+const overlayStyle = computed(() => {
+  const buildDndState = runtimeContext.buildDndState.value
+  if (buildDndState) {
+    const buildContentWindowInner = document.getElementById('build-content-window-inner')
+    const hoverElement = document.getElementById(buildDndState.hoverCmpId)
+
+    if (buildContentWindowInner && hoverElement) {
+      const {
+        hoverSelf,
+        hoverTop,
+        hoverRight,
+        hoverBottom,
+        hoverLeft,
+        hoverCmpParentIsRow,
+      } = buildDndState
+
+      const { top: buildContentWindowTop, left: buildContentWindowLeft } =
+        buildContentWindowInner.getBoundingClientRect()
+
+      const {
+        top: elementTop,
+        right: elementRight,
+        bottom: elementBottom,
+        left: elementLeft,
+        width: elementWidth,
+        height: elementHeight,
+      } = hoverElement.getBoundingClientRect()
+
+      const builderScale = editor.value?.builderScale ?? 1
+      const outlineThickness = Math.min(
+        hoverOutlineThickness,
+        hoverOutlineThickness * builderScale,
+      )
+
+      const elementRelativeTop = elementTop - buildContentWindowTop
+      const elementRelativeRight = elementRight - buildContentWindowLeft
+      const elementRelativeBottom = elementBottom - buildContentWindowTop
+      const elementRelativeLeft = elementLeft - buildContentWindowLeft
+
+      let [overlayTop, overlayLeft, overlayWidth, overlayHeight] = [0, 0, 0, 0]
+      let [overlayBackgroundColor, overlayBorder] = ['', '']
+
+      overlayTop = elementRelativeTop
```

Comment:
I factored these out to help de-clutter a little, I don't think it breaks anything but you might want to double check!

Reply:
Hi @sampullman 

I think everything still works after the refactor! One nitpick is that we could probably change this to `let [overlayTop, overlayLeft, overlayWidth, overlayHeight] = [elementRelativeTop, elementRelativeLeft, 0, 0]`, but that doesn't really make anything better, though! 🤔

Reply:
That makes sense! But I think it will cause prettier to expand the statement over 5 lines 😬